File: frontend/src/main/java/com/innertrack/model/Notification.java
package com.innertrack.model;

import java.time.LocalDateTime;

public class Notification {
    private int id;
    private int userId;
    private String type;        // CONTACT_REQUEST | MESSAGE | SYSTEM
    private String title;
    private String body;
    private boolean isRead;
    private Integer referenceId; // e.g. contact_request.id or conversation.id
    private LocalDateTime createdAt;

    public Notification() {}

    public Notification(int userId, String type, String title, String body, Integer referenceId) {
        this.userId = userId;
        this.type = type;
        this.title = title;
        this.body = body;
        this.referenceId = referenceId;
        this.isRead = false;
    }

    public int getId()                           { return id; }
    public void setId(int id)                    { this.id = id; }
    public int getUserId()                       { return userId; }
    public void setUserId(int v)                 { this.userId = v; }
    public String getType()                      { return type; }
    public void setType(String type)             { this.type = type; }
    public String getTitle()                     { return title; }
    public void setTitle(String title)           { this.title = title; }
    public String getBody()                      { return body; }
    public void setBody(String body)             { this.body = body; }
    public boolean isRead()                      { return isRead; }
    public void setRead(boolean read)            { isRead = read; }
    public Integer getReferenceId()              { return referenceId; }
    public void setReferenceId(Integer v)        { this.referenceId = v; }
    public LocalDateTime getCreatedAt()          { return createdAt; }
    public void setCreatedAt(LocalDateTime v)    { this.createdAt = v; }
}

File: backend/src/Entity/Notification.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'notification')]
class Notification
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id', nullable: false)]
    private User $user;

    #[ORM\Column(length: 30, options: ['default' => 'SYSTEM'])]
    private string $type = 'SYSTEM';

    #[ORM\Column(length: 150)]
    private string $title;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $body = null;

    #[ORM\Column(name: 'is_read', type: 'boolean', options: ['default' => false])]
    private bool $isRead = false;

    #[ORM\Column(name: 'reference_id', nullable: true)]
    private ?int $referenceId = null;

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private \DateTimeInterface $createdAt;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getUser(): User { return $this->user; }
    public function setUser(User $user): self { $this->user = $user; return $this; }
    public function getType(): string { return $this->type; }
    public function setType(string $type): self { $this->type = $type; return $this; }
    public function getTitle(): string { return $this->title; }
    public function setTitle(string $title): self { $this->title = $title; return $this; }
    public function getBody(): ?string { return $this->body; }
    public function setBody(?string $body): self { $this->body = $body; return $this; }
    public function isRead(): bool { return $this->isRead; }
    public function setIsRead(bool $isRead): self { $this->isRead = $isRead; return $this; }
    public function getReferenceId(): ?int { return $this->referenceId; }
    public function setReferenceId(?int $referenceId): self { $this->referenceId = $referenceId; return $this; }
    public function getCreatedAt(): \DateTimeInterface { return $this->createdAt; }

    public function getIcon(): string
    {
        return match($this->type) {
            'MESSAGE' => '💬',
            'CONTACT_REQUEST' => '🤝',
            'SYSTEM' => '⚙️',
            default => '🔔',
        };
    }
}
